saleor/order/utils.py

```diff
@@ -212,3 +215,36 @@ def restock_fulfillment_lines(fulfillment):
         if line.order_line.variant and line.order_line.variant.track_inventory:
             increase_stock(
                 line.order_line.variant, line.quantity, allocate=True)
+
+
+def validate_shipping_method(order):
+    """Validates `order.shipping_method`.
+
+    Clears shipping attributes for a draft order if its `shipping_method`
+    is invalid after the draft order has been updated. A draft order is invalid
+    if either its price or weight is not within the range of the price or
+    weight of the `ShippingMethod` that the user has selected.
+    """
+    if order.is_shipping_required():
+        valid_methods = (
+            ShippingMethod.objects.applicable_shipping_methods(
+                price=order.total.gross.amount,
+                weight=order.get_total_weight(),
+                country_code=order.shipping_address.country.code))
+        valid_methods = valid_methods.values_list('id', flat=True)
+        if order.shipping_method.pk not in valid_methods:
+            clear_shipping_attributes(order)
+    elif order.shipping_method and not order.is_shipping_required():
+        clear_shipping_attributes(order)
+
+
+def clear_shipping_attributes(order):
+    order.shipping_method = None
+    order.shipping_method_name = None
+    order.shipping_price = TaxedMoney(
+        net=Money(0, settings.DEFAULT_CURRENCY),
+        gross=Money(0, settings.DEFAULT_CURRENCY))
+    order.save(
+        update_fields=[
+            'shipping_method', 'shipping_method_name',
+            'shipping_price'])
```

Comment:
That actually won't work, as `shipping_price` is a proxy field for `shipping_price_net` and `shipping_price_gross`, which should be listed instead.

Reply:
```
order.shipping_price_net = Money(0, settings.DEFAULT_CURRENCY)
order.shipping_price_gross = Money(0, settings.DEFAULT_CURRENCY)
```
This?

Reply:
You can assign normally, as you did, to the `shipping_price`, but in `update_fields` you should list both of the fields (`shipping_price_net` and `shipping_price_gross`) separately.